src/transformers/models/moshi/convert_moshi_transformers.py

```diff
@@ -0,0 +1,331 @@
+
+    audio_checkpoint = mimi_model.state_dict()
+    original_checkpoint.update({f"audio_encoder.{key}": value for (key, value) in audio_checkpoint.items()})
+
+    model = _convert_model(original_checkpoint, model, convert_list, device, config)
+
+    model.save_pretrained(pytorch_dump_folder_path)
+
+    if repo_id:
+        print("Pushing to the hub...")
+        model.push_to_hub(repo_id)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--checkpoint_path", required=True, default=None, type=str, help="Path to original checkpoint")
+    parser.add_argument(
+        "--tokenizer_vocab_path", required=False, default=None, type=str, help="Path to original tokenizer vocab file"
+    )
+    parser.add_argument("--mimi_repo_id", required=True, default=None, type=str, help="Repository id to HF Mimi.")
+    parser.add_argument("--config_path", default=None, type=str, help="Path to hf config.json of model to convert")
+    parser.add_argument(
+        "--pytorch_dump_folder_path", required=True, default=None, type=str, help="Path to the output PyTorch model."
+    )
+    parser.add_argument(
+        "--push_to_hub", default=None, type=str, help="Where to upload the converted model on the 🤗 hub."
+    )
+    parser.add_argument("--test_tokenizer", default=None, type=bool, help="Test the tokenizer on hard sentences.")
+
+    args = parser.parse_args()
+
+    # convert tokenizer
+    if args.tokenizer_vocab_path:
+        original_tokenizer = sentencepiece.SentencePieceProcessor(args.tokenizer_vocab_path)
+        tokenizer = MoshiConverter(args.tokenizer_vocab_path).converted()
+        tokenizer = PreTrainedTokenizerFast(
+            tokenizer_object=tokenizer,
+            chat_template=None,
+            unk_token="<unk>",
+            model_input_names=["input_ids", "attention_mask"],
+            clean_up_tokenization_spaces=False,
+            bos_token_id=original_tokenizer.bos_id(),
+            eos_token_id=original_tokenizer.eos_id(),
+            pad_token_id=original_tokenizer.pad_id(),
+        )
+        if args.test_tokenizer:
+            import tqdm
+            from datasets import load_dataset
+
+            dataset = load_dataset("google/code_x_glue_ct_code_to_text", "go")
+            for item in tqdm.tqdm(dataset["validation"]):
+                string = item["code"]
+                encoded1 = tokenizer.encode(string)
+                encoded2 = original_tokenizer.encode(string)
+
+                decoded1 = tokenizer.decode(encoded1)
+                decoded2 = original_tokenizer.decode(encoded1)
+
+                if decoded1 != decoded2:
+                    raise ValueError(
+                        "Hint: the following tokenization.decode diff were obtained for HF tokenizer vs original:\n "
+                        f"elements in HF: {set(decoded1)-set(decoded2)} \nvs\n "
+                        f"elements in original: {set(decoded2)-set(decoded1)} \n\n{string}"
```

Comment:
Not sure we want to scare people off, let's remove this and test this in the tests!